source/allocator.c

```diff
@@ -117,6 +117,57 @@ struct aws_allocator *aws_default_allocator(void) {
     return &default_allocator;
 }
 
+static void *s_no_align_malloc(struct aws_allocator *allocator, size_t size) {
+    (void)allocator;
+    void *result = malloc(size);
```

Comment:
```suggestion
    /* Protect against https://wiki.sei.cmu.edu/confluence/display/c/MEM04-C.+Beware+of+zero-length+allocations */
    AWS_FATAL_PRECONDITION(size > 0);
    void *result = malloc(size);
```

Reply:
we have that check in the aws_mem_acquire already